### ISSUE TITLE
Sound only works with basic.sh, not virt-manager

### ISSUE THREAD

**Issue by geckolinux** [foxlet/macOS-Simple-KVM#57]
Hi there,

Can anybody confirm or deny if sound (output) works with virt-manager? I can only get it work with `basic.sh`. There appears to be some difference in the virt-manager template.xml that `make.sh` generates. In the template.xml there is:

````````````
    <controller type='usb' index='0' model='ich9-ehci1'>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1d' function='0x7'/>
    </controller>
    <controller type='usb' index='0' model='ich9-uhci1'>
      <master startport='0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1d' function='0x0' multifunction='on'/>
    </controller>
    <controller type='usb' index='0' model='ich9-uhci2'>
      <master startport='2'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1d' function='0x1'/>
    </controller>
    <controller type='usb' index='0' model='ich9-uhci3'>
      <master startport='4'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x1d' function='0x2'/>
    </controller>
````````````

Whereas `basic.sh` uses `-device ich9-ahci,id=sata`

Thanks!

**Comment by foxlet** [comment]
Those are the hard disk and USB controllers, nothing to do with sound.

**Comment by idnovic** [comment]
I need to reinstall the vm but as far as i remember sound did not work for me

**Comment by geckolinux** [comment]
Hm, OK. Well for me, sound output worked great (microphone didn't work) with `basic.sh`, but no sound devices under virt-manager.

**Comment by idnovic** [comment]
So I reinstalled macOS 10.13 and sound did work under virt-manager. But Since i found out that 10.14 just started supporting virt-io I deleted the vm again and I am doing the virt-io setup now :)

**Comment by IronicBadger** [comment]
@idnovic could you elaborate on 'supporting virt-io'? I'm quite familiar with the process of virtio drivers in Winbloze and Linux.

**Comment by idnovic** [comment]
> @idnovic could you elaborate on 'supporting virt-io'? I'm quite familiar with the process of virtio drivers in Winbloze and Linux.

Well sind around 10.14.3 macOS includes some virt-io drivers. I can tell you that the qemu-std gpu driver works. Virt-io for block devices does not work for me. Ethernet is still e1000.

https://www.kraxel.org/blog/2019/06/macos-qemu-guest/
https://github.com/foxlet/macOS-Simple-KVM/blob/master/virtio.sh

**Comment by idnovic** [comment]
about the sound issue:
sound did work for me on 10.13
sound does not work for me on 10.14
both virt-manager

**Comment by foxlet** [comment]
virtIO can only be used for existing installations, the Installer environment doesn't include Block drivers, only the final install.

**Comment by idnovic** [comment]
> virtIO can only be used for existing installations, the Installer environment doesn't include Block drivers, only the final install.

I can not boot with virtio after install. I just get the grey x at around 50%

**Comment by foxlet** [comment]
Gray X? There should be no graphics but verbose messages if you're using the proper ESP.

**Comment by idnovic** [comment]
> Gray X? There should be no graphics but verbose messages if you're using the proper ESP.

Yes. I deactivated the verbose output because the vm boots faster without it. It does work with the sata driver. Just not with virtio drive. 

macOS 10.14 installed with apfs even thought I did not choose apfs. I will just wait until virtio is more useable with macOS.

If you want the information I can boot with -v and tell you the problem with virtio hard drive.

**Comment by kode54** [comment]
It could be the fact that libvirtd sets it up with an `hda-duplex` device, while `basic.sh` pairs it with a `hda-output` device.

**Comment by eliasailenei** [comment]
Sound does work on Virtual Machine Manager and i can show you how. I actually found this out after many days researching with no solution. First, re install mac OS high sierra and don't put the GPU yet. Once done install, put GPU and DO NOT REMOVE THE SPICE YET. Install the driver by putting this in the terminal.
`bash <(curl -s https://raw.githubusercontent.com/Benjamin-Dobell/nvidia-update/master/nvidia-update.sh)`

Then shutdown don't restart. 

Remove the spice, and then launch mac os for the first time. 

Shutdown and then add the spice manually and don't modify them. Just modify audio to be ich6.

That's it! It would prioritize your GPU and you have audio.

Only bad thing is that it is from Linux not macOS but audio is audio unless your deaf lmao.
